app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/processor/TransactionTypeExtensions.kt

```diff
@@ -690,40 +668,88 @@ fun List<Transferable>.toAccountWithTransferableResources(
     }
 }
 
+suspend fun ExecutionSummary.allOwnedAccounts(getProfileUseCase: GetProfileUseCase): List<Network.Account> {
+    val involvedAccountAddresses = accountWithdraws.keys + accountDeposits.keys
+    return getProfileUseCase.accountsOnCurrentNetwork().filter {
+        involvedAccountAddresses.contains(it.address)
+    }
+}
+
 fun ExecutionSummary.toWithdrawingAccountsWithTransferableAssets(
     involvedAssets: List<Asset>,
-    allOwnedAccounts: List<Network.Account>
-) = accountWithdraws.map { withdrawEntry ->
-    withdrawEntry.value.map { resource ->
-        if (resource.isNewlyCreated(summary = this)) {
-            resource.toNewlyCreatedTransferableAsset(resource.newlyCreatedMetadata(summary = this))
+    allOwnedAccounts: List<Network.Account>,
+    aggregateByResourceAddress: Boolean = false // for now I turned aggregation off to not introduce any bugs
+): List<AccountWithTransferableResources> {
+    return accountWithdraws.map { withdrawEntry ->
+        if (aggregateByResourceAddress) {
+            withdrawEntry.value.groupBy { it.resourceAddress }.map { indicatorsPerResource ->
+                val resources = indicatorsPerResource.value
+                when (val first = resources.first()) {
```

Comment:
minor: val `resource` since it returns a resource and then you check the type of resource.